packages/webdriverio/src/utils/Timer.ts

```diff
@@ -88,23 +96,27 @@ class Timer {
         }
 
         result.then(
-            (res) => this.checkCondition(null, res),
-            (err) => this.checkCondition(err)
+            (res: any) => this.checkCondition(null, res),
+            (err: any) => this.checkCondition(err)
```

Comment:
```suggestion
            (err: Error) => this.checkCondition(err)
```

Reply:
Here actually if we get the type and the condition right in:
```ts
const result = this._fn() as any

if (typeof result.then !== 'function') { //...
```
Then these two `any`'s will be correctly inferred